src/data_capture/video_capture/scripts/capture.py

```diff
@@ -9,6 +9,76 @@
 
 from video_recorder import VideoRecorder
 
+from threading import Timer
+import os
+from time import sleep
+
+import psutil
+
+
+"""
+This is a threaded timer to execute a piece of code periodically 
+without blocking the main execution.
+
+This non-blocking method allows the code to keep running while also
+executing our function every n seconds
+
+NOTE:
+- start() and stop() are safe to call multiple times even if the timer 
+has already started/stopped
+
+- You can change interval anytime, it will be effective after next run. 
+Same for args, kwargs and even function!
+"""
+
+class RepeatedTimer(object):
+    def __init__(self, interval, function, *args, **kwargs):
+        self._timer     = None
+        self.interval   = interval
+        self.function   = function
+        self.args       = args
+        self.kwargs     = kwargs
+        self.is_running = False
+        self.start()
+
+    def _run(self):
+        try:
+            self.is_running = False
+            self.start()
+            self.function(*self.args, **self.kwargs)
+        except Exception as e:
+            if 'rt' in locals() or 'rt' in globals():
+                rt.stop()
+            rospy.logerr(e)
+
+
+    def start(self):
+        if not self.is_running:
+            self._timer = Timer(self.interval, self._run)
+            self._timer.start()
+            self.is_running = True
+
+    def stop(self):
+        self._timer.cancel()
+        self.is_running = False
+
+"""
+This function is used to monitor the memory utilization and throw an error 
+if it exceeds a preset value.
+
+Arguments:
+memory_upperlimit_percent: The upperlimit of the memory utilization (float)
+"""
+def monitorAvailableMemory(memory_upperlimit_percent, video_recorder_object):
+    # Utilized memory
+    utilized_memory = psutil.virtual_memory().percent
+    
+    if utilized_memory > memory_upperlimit_percent:
```

Comment:
Any reason not to do `if cond1 and cond2`? You can put these both in a function that returns a bool and expresses what the if statement is checking.